index.html

```diff
@@ -484,14 +488,146 @@ <h4><code>readproperty</code></h4>
+          <thead>
+            <tr>
+              <th>Member</th>
+              <th>Type</th>
+              <th>Assignment</th>
+              <th>Description</th>
+            </tr>
+          </thead>
+          <tbody>
+            <tr>
+              <td><code>messageType</code></td>
+              <td>string</td>
+              <td>"response"</td>
+              <td>A string which denotes that this message is a response sent from a <a>Thing</a> to a <a>Consumer</a>.
+              </td>
+            </tr>
+            <tr>
+              <td><code>operation</code></td>
+              <td>string</td>
+              <td>"writeproperty"</td>
+              <td>A string which denotes that this message relates to a <code>writeproperty</code> operation.
+              </td>
+            </tr>
+            <tr>
+              <td><code>name</code></td>
+              <td>string</td>
+              <td>Mandatory</td>
+              <td>The name of the <a>Property</a> being written, as per its key in the <code>properties</code> member of
+                the
+                <a>Thing Description</a>.
+              </td>
+            </tr>
+            <tr>
+              <td><code>value</code></td>
+              <td>any</td>
+              <td>Optional</td>
+              <td>The value which has been successfully assigned to the <a>Property</a> being written, with a format
+                conforming to the data schema of the corresponding PropertyAffordance in the <a>Thing Description</a>.
+              </td>
+            </tr>
+            <tr>
+              <td><code>timestamp</code></td>
+              <td>string</td>
+              <td>Optional</td>
+              <td>A timestamp in date-time format [[RFC3339]] set to the time the property write took place.</td>
+            </tr>
+          </tbody>
+        </table>
+        <pre class="example" title="writeproperty response message">
+          {
+            "thingID": "https://mythingserver.com/things/mylamp1",
+            "messageID": "db25fe4f-bee8-43a7-8ff0-3a1ff6e620b0",
+            "messageType": "response",
+            "operation": "writeproperty",
+            "name": "on",
+            "value": true,
+            "timestamp": "2024-01-13T23:20:50.52Z",
+            "correlationID": "f6cf46a8-9c96-437e-8b53-925b7679a990"
+          }
+        </pre>
+        <p>If the <a>Thing</a> can confirm that the requested value has been set successfully then the response message
+          to a <code>writeproperty</code> request SHOULD contain a <code>value</code> member with its value set to the
+          value which has been successfully set.
+        </p>
+        <p>If the <a>Thing</a> can not confirm that the requested value has been set successfully (e.g. in the case of a
+          write-only property) then the response message to a <code>writeproperty</code> request MUST NOT contain a
+          <code>value</code> member.
+        </p>
+      </section>
```

Comment:
What about the use-case if a (battery operated) device is asleep and property writes are delayed until it wakes up.  In this case an intermediary (for example gateway, zwave controller) will queue the request and no value is returned. Is it useful to somehow indicate that the request is still pending. It is very similar to long running actions. This might be out of scope of this protocol binding but still good to know how this is supposed to be handled.

Reply:
I guess this comes under "If the Thing can not confirm that the requested value has been set successfully", so the Thing would send a response message to confirm that the write has been requested, but omit the value member to indicate to indicate to the Consumer that it may not (yet) have taken effect. A Consumer can do a `readproperty` operation later on to confirm whether the value change was successful, or monitor the device with `observeproperty` until the change takes effect. Does that make sense?

Reply:
I've added that as an example in the assertion, thanks.